### ISSUE TITLE
Clarification Needed on Target Task Data Splits for Fine-Tuning and Evaluation

### ISSUE THREAD

**Issue by yoyowang0109** [som-shahlab/femr#251]
Hi, thank you for your work on this project.

I'm currently working on implementing the target tasks and noticed that the documentation does not specify how the data is split for fine-tuning and evaluation on the target tasks. Specifically, it's unclear what portion of the target task data is used for fine-tuning and what is reserved for final inference/evaluation.

For reference, I’ve read Appendix B of the MOTOR paper regarding data processing. From what I understand, the validation set mentioned there is used for hyperparameter tuning during pretraining, not for the target tasks. If that’s correct, it seems only a testing set is used for evaluating the target tasks, and the fine-tuning split is not described.

Could you please clarify the data splitting protocol used for the target tasks (i.e., fine-tuning vs. evaluation)?

Thank you for your help!

**Comment by EthanSteinberg** [comment]
For the MOTOR paper, the splits used for target tasks matched the splits used for pretraining. 15% of patients were used for final evaluation, 15% were used for hyperparameter tuning, and 70% were used for fine-tuning.

Note that the main difference is that the target task datasets are a subset of the main dataset. Compare table S4 with table S2. But the splits are maintained during that subsetting process. So if a patient was in the big evaluation set, that same patient would be in a particular target task evaluation set.

I'm going to close this ticket, but feel free to reopen if you have further questions!

**Comment by yoyowang0109** [comment]
@EthanSteinberg, thank you for your clarification. However, I found that the sum of the number of patients listed in Table S4 does not match any entry in Table S2. Take the STARR-OMOP dataset, for instance, the sum in Table S4 is 227,865, which is less than the number of patients in the test set in Table S2 (409,820). Moreover, I found that the number of controls is 4 times that of cases for every target task in Table S4. So I wonder whether sampling is performed on the test set to select patients for subsequent target tasks.

**Comment by EthanSteinberg** [comment]
@yoyowang0109  The numbers in S4 are the total patients per task, not the validation set, so including training, validation and test. 

In addition, you would not expect the sums in S4 to match the total population sums in S2 for two reasons.

First, there is overlap between patients in the task sets. This is because a patient can have multiple conditions and of course the controls will have some overlap. 

Second, as described in Appendix D, there is subsampling of controls going on where 4 controls are being sampled for each case.

**Comment by yoyowang0109** [comment]
@EthanSteinberg, thank you for clarifying. I'd like to confirm my understanding of the data split in MOTOR—please let me know if I’ve misunderstood anything.

From what I gather, the dataset is first split into training (70%), validation (15%), and test (15%) sets. During the pre-training stage, the training set is used across various tasks (excluding the target tasks), and the validation set is used for hyperparameter tuning. In the fine-tuning stage, the same training and validation sets are used, but this time specifically for the target tasks. Finally, the test set is used solely for inference or evaluation on the target tasks.

**Comment by EthanSteinberg** [comment]
@yoyowang0109  Yes, that is correct.
